### PR TITLE
fix(gatsby): Adapter header rules

### PR DESCRIPTION
**Pull request by kathmbeck**
## Description

Adjustment to reduce the header file size. 

Based off of work done by https://github.com/gatsbyjs/gatsby/pull/38561. The primary difference is the equality checks for filtering the headers. `includes` doesn't work since JS checks for reference equality. 

### Tests

Unit tests 

## Related Issues

Fixes https://linear.app/netlify/issue/FRA-17/reduce-headers-file-size

### CODE REVIEW

File: packages/gatsby/src/utils/adapter/__tests__/__snapshots__/manager.ts.snap [hunk 1]
```diff
@@ -146,7 +146,7 @@ Array [
   },
   Object {
     "functionId": "static-index-js",
-    "path": "/api/static/",
+    "path": "/api/static",
```
Comment (pieh) [inline packages/gatsby/src/utils/adapter/__tests__/__snapshots__/manager.ts.snap:149]: Seems like those snapshot changes are incorrect ( https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/91766/workflows/85a3583b-de8f-486a-a473-70b21d1cc4ad/jobs/1111387/tests )

Comment (kathmbeck) [reply packages/gatsby/src/utils/adapter/__tests__/__snapshots__/manager.ts.snap:149]: The snapshots were passing locally at my EOD, so I was going to take a closer look today at the cause here.

**Comment by kathmbeck** [comment]
Tests were passing, but a new test is failing after merging in master https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/91772/workflows/deb2a349-7db6-4970-965d-01578ec83ea5/jobs/1111499/tests

**Comment by kathmbeck** [comment]
> Tests were passing, but a new test is failing after merging in master https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/91772/workflows/deb2a349-7db6-4970-965d-01578ec83ea5/jobs/1111499/tests

Ran again and it passed 🎉

**Review by mraerino — APPROVED** [review]
looks good to me!
did you want to add an integration test with a dummy adapter maybe?

**Comment by kathmbeck** [comment]
> looks good to me! did you want to add an integration test with a dummy adapter maybe?

As discussed in the sync, I'll update the E2E tests in the follow-up PR with the adapter changes.

**Pull request merged**